### DISCUSSION TITLE
Use environment variables in plugin configuration

### DISCUSSION THREAD

**Discussion by djeanprost** [semantic-release/semantic-release#2785] [Configuration] [answered]
Hello,

I want to use @terrestris/maven-semantic-release to release maven artifact from a gitlab pipeline.
This plugin requires to provide maven settings xml file. The location of this file is contained in a gitlab variable.
I would like to be able to use this env variable in my plugin config, but I cannot find any documentation about the use of environment variable for `.releaserc.json` file
how can I do ?
thank you

```
{
  "tagFormat": "${version}",
  "plugins": [
    "@semantic-release/commit-analyzer",
    ...,
    "@semantic-release/gitlab",
    ["@terrestris/maven-semantic-release" ,{
        "settingsPath": "${MY_ENV_VARIABLE}",
        "clean": "true",
        "mavenTarget": "deploy",
        "updateSnapshotVersion": "true",
        "processAllModules": "true"
    }]
  ]
}

```

**Comment by travi** [comment, marked as answer]
the configuration file [can be defined as a javascript file](https://semantic-release.gitbook.io/semantic-release/usage/configuration#configuration-file), which should enable you to resolve environment variables

**Reply by djeanprost** [reply]
Hello @travi 

Thank you for your answer. As I'm not really familiar with javascript, would you mind providing a example of such configuration file please ?

**Reply by djeanprost** [reply]
Well ChatGPT dit the trick.
Thank you @travi for providing the initial answer.

```
module.exports = {
  branches: ["main"],
  plugins: [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    [
      "@terrestris/maven-semantic-release",
      {
        url: "https://your-maven-repository-url.com",
        repositoryId: "your-repository-id",
        snapshotRepositoryId: "your-snapshot-repository-id",
        groupId: "com.example",
        scmId: "git",
        scmUrl: "https://github.com/your-username/your-repo.git",
        settingsXmlPath: process.env.MAVEN_SETTINGS_XML_PATH,
      },
    ],
    [
      "@semantic-release/git",
      {
        assets: ["pom.xml", "src/**/*", "CHANGELOG.md"],
        message: "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}",
      },
    ],
  ],
};
```
